app/src/main/java/org/mozilla/fenix/settings/advanced/LocaleViewHolders.kt

```diff
@@ -21,15 +21,226 @@ class LocaleViewHolder(
 
     override fun bind(locale: Locale) {
         // Capitalisation is done using the rules of the appropriate locale (endonym and exonym).
-        locale_title_text.text = locale.getDisplayName(locale).capitalize(locale)
+        locale_title_text.text = getDisplayName(locale)
         // Show the given locale using the device locale for the subtitle.
-        locale_subtitle_text.text = locale.displayName.capitalize(Locale.getDefault())
+        locale_subtitle_text.text = locale.getProperDisplayName()
         locale_selected_icon.isVisible = isCurrentLocaleSelected(locale, isDefault = false)
 
         itemView.setOnClickListener {
             interactor.onLocaleSelected(locale)
         }
     }
+
+    private fun getDisplayName(locale: Locale): String {
+        val displayName = locale.getDisplayName(locale).capitalize(locale)
+        if (displayName.equals(locale.toString(), ignoreCase = true)) {
+            return LOCALE_TO_DISPLAY_NATIVE_NAME_MAP[locale.toString()] ?: displayName
+        }
+        return displayName
+    }
+
+    @SuppressWarnings("LargeClass")
+    companion object {
+        val LOCALE_TO_DISPLAY_NATIVE_NAME_MAP: Map<String, String> = mapOf(
```

Comment:
Hmm I'm not sure that I like having hard-coded names like this in the ViewHolder. I believe there should be a way to handle this using [Locale](https://developer.android.com/reference/java/util/Locale) or [LocaleList](https://developer.android.com/reference/android/os/LocaleList), but I think we should have this logic located in Android Components in the [LocaleManager.kt](https://github.com/mozilla-mobile/android-components/blob/master/components/support/locale/src/main/java/mozilla/components/support/locale/LocaleManager.kt)

Reply:
What do you think? 🤔

Reply:
@mcarare does Focus has a similar logic? If so, I think it would be a good idea to have the code on AC to de-duplicate the logic from both source codes, this way we will has a single source of truth. If we can't address it as part of this pr we could add reference ticket for addressing it on follow-up task. Either here or on AC it will be good to add test for covering this new behavior.

Reply:
@Amejia481  I am not sure, 🤔 . There already is an extension in AC `Locale.getLocaleTag()` but I am not sure what it is used for. I still do not think Locale Manager should handle logic outside of setting or getting the system locale, meaning logic that describes behavior. IMHO display logic should be left to the client, or at best, handled separately from "management" logic.

Reply:
I was more concern about duplicating logic that other apps could be using as well, if this is the case we could put it in any place on ac as long as apps can share the functionality, but as you mentioned below this could be addressed in separate ticket, just let add some tests to cover the new behavior. :)

Reply:
Apart from that it's OK for me to land.

Reply:
@Amejia481 I added a couple of tests and confirmed that the results are OS-dependent. I added a comment so we can revise the tests when we can use SDK 30 in unit tests.

For example, if I run the app on a device, the display name for `vec` locale is "Venitian" for SDK 30 and that is the value displayed on the device (It actually might be wrong because the native language name is "Vèneto", but that is a L10n discussion). That would mean that when run on SDK 30 the test would fail. 
The actual result when running the test with SDK 28 is that the display name for `vec` locale is "vec" and the display value uses the fallback value of "Vèneto" and the test passes.

Reply:
👍🏽